No way to set build config memory/CPU request and limi

#1604 # Description
Could you please add a way to set resources configuration for buildConfig like

```
apiVersion: "v1"
kind: "BuildConfig"
metadata:
  name: "sample-build"
spec:
  resources:
    limits:
      cpu: "100m" 
      memory: "256Mi"
```

### Info
I'm using openshift, my build is OOM killed due to 250Mi memory limit ranges

```
spec:
  limits:
    - type: Pod
      max:
        cpu: '4'
        memory: 4Gi
      min:
        cpu: 10m
        memory: 3Mi
    - type: Container
      max:
        cpu: '4'
        memory: 4Gi
      min:
        cpu: 10m
        memory: 3Mi
      default:
        cpu: 100m
        memory: 250Mi
      defaultRequest:
        cpu: 10m
        memory: 10Mi
```

Not seems to work

```xml
                    <plugin>
                        <groupId>io.fabric8</groupId>
                        <artifactId>fabric8-maven-plugin</artifactId>
                        <version>4.4.0</version>
                        <configuration>
                            </generator>
                            <resources>
                                <openshiftBuildConfig>
                                    <limits>
                                        <cpu>100m</cpu>
                                        <memory>1Gi</memory>
                                    </limits>
                                </openshiftBuildConfig>
                            </resources>
```


```console
[ERROR] Failed to execute goal io.fabric8:fabric8-maven-plugin:4.4.0:resource (default) on project web: Unable to parse configuration of mojo io.fabric8:fabric8-maven-plugin:4.4.0:resource for parameter openshiftBuildConfig: Cannot create instance of class io.fabric8.maven.core.config.OpenshiftBuildConfig: io.fabric8.maven.core.config.OpenshiftBuildConfig.<init>() -> [Help 1]
3435 org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.fabric8:fabric8-maven-plugin:4.4.0:resource (default) on project web: Unable to parse configuration of mojo io.fabric8:fabric8-maven-plugin:4.4.0:resource for parameter openshiftBuildConfig: Cannot create instance of class io.fabric8.maven.core.config.OpenshiftBuildConfig
3436     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
3437     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
3438     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
3439     at org.apache.maven.lifecycle.internal.MojoExecutor.executeForkedExecutions (MojoExecutor.java:355)
3440     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:200)
3441     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
3442     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
3443     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
3444     at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:190)
3445     at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:186)
3446     at java.util.concurrent.FutureTask.run (FutureTask.java:264)
3447     at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:515)
3448     at java.util.concurrent.FutureTask.run (FutureTask.java:264)
3449     at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1128)
3450     at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:628)
3451     at java.lang.Thread.run (Thread.java:834)
3452 Caused by: org.apache.maven.plugin.PluginConfigurationException: Unable to parse configuration of mojo io.fabric8:fabric8-maven-plugin:4.4.0:resource for parameter openshiftBuildConfig: Cannot create instance of class io.fabric8.maven.core.config.OpenshiftBuildConfig
3453     at org.apache.maven.plugin.internal.DefaultMavenPluginManager.populatePluginFields (DefaultMavenPluginManager.java:665)
3454     at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getConfiguredMojo (DefaultMavenPluginManager.java:597)
3455     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:124)
3456     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
3457     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
3458     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
3459     at org.apache.maven.lifecycle.internal.MojoExecutor.executeForkedExecutions (MojoExecutor.java:355)
3460     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:200)
3461     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
3462     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
3463     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
3464     at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:190)
3465     at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:186)
3466     at java.util.concurrent.FutureTask.run (FutureTask.java:264)
3467     at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:515)
3468     at java.util.concurrent.FutureTask.run (FutureTask.java:264)
3469     at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1128)
3470     at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:628)
3471     at java.lang.Thread.run (Thread.java:834)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,7 +10,20 @@ We use semantic versioning in some slight variation until our feature set has st
 
 After this we will switch probably to real [Semantic Versioning 2.0.0](http://semver.org/)
 
+## Extracting changelog portions
+
+We provide a script to extract changelog portions and automatic link building to send notifications
+(i.e. e-mail) about new releases
+([scripts/extract-changelog-for-version.sh](https://github.com/fabric8io/fabric8-maven-plugin/blob/master/scripts/extract-changelog-for-version.sh))
+
+Usage:
+```
+# ./scripts/extract-changelog-for-version.sh semanticVersionNumber [linkLabelStartNumber]
+./scripts/extract-changelog-for-version.sh 1.3.37 5
+```
+
 ### 4.5-SNAPSHOT
+* Fix #1789: script to extract changelog information for notifications
 
 ### 4.4.0 (2020-02-13)
 * Fix #1572: Support maven --batch-mode option

--- a/scripts/extract-changelog-for-version.sh
+++ b/scripts/extract-changelog-for-version.sh
@@ -1,0 +1,77 @@
+#!/bin/bash
+#
+# Copyright 2016 Red Hat, Inc.
+#
+# Red Hat licenses this file to you under the Apache License, version
+# 2.0 (the "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.  See the License for the specific language governing
+# permissions and limitations under the License.
+#
+
+trap 'exit' ERR
+
+START_LINK=10
+BASEDIR=$(dirname "$BASH_SOURCE")
+
+function checkInput() {
+  if [ "$#" -lt 1 ]; then
+    echo -e "This script extracts chagnelog version contents from CHANGELOG.md"
+    echo -e "Usage: ./extract-changelog-for-version.sh semVer [startLinkNumber]\n"
+    echo -e "Must set a valid semantic version number (e.g. 1.3.37)"
+    exit 1;
+  fi
+  dotCount=$(echo "$1" | tr -d -c '.' | wc -c)
+  if [ "$dotCount" -ne 2 ]; then
+      echo "Provided version has an invalid format, should be semver compliant (e.g. 1.3.37)"
+      exit 1;
+  fi
+}
+
+function extractChangelogPortion() {
+  sed -e "/### ""$1""/,/###/!d" "$BASEDIR/../CHANGELOG.md"
+}
+
+function removeLastLine() {
+  echo "$1" | sed '$d'
+}
+
+function replaceBullets() {
+  echo -e "$1" | sed -e "s/^*/-/"
+}
+
+function addLinks() {
+  lines=""
+  links=""
+  currentLink="$START_LINK"
+  if [ -n "$2" ]; then currentLink="$2" ; fi
+  while read -r line; do
+    issueNumber=$(echo "$line" | sed -En 's/.*?#([0-9]+).*/\1/p')
+    if [ -z "$issueNumber" ]; then
+      lines+="$line\n";
+    else
+      lines+="$line [$currentLink]\n"
+      links+="[$currentLink]: https://github.com/fabric8io/fabric8-maven-plugin/issues/$issueNumber\n"
+      currentLink=$((currentLink + 1));
+    fi
+  done < <(echo "$1")
+  echo -e "$lines\n$links";
+}
+
+function processChangelog() {
+  changelog=$1
+  changelog=$(extractChangelogPortion "$changelog")
+  changelog=$(removeLastLine "$changelog")
+  changelog=$(replaceBullets "$changelog")
+  changelog=$(addLinks "$changelog" "$2")
+  echo "$changelog";
+}
+
+checkInput "$@"
+processChangelog "$@"

